where.ts

```diff
@@ -0,0 +1,14 @@
+
+import curryN from "./utils/curry_n.ts"
+import { Curry2 } from "./utils/types.ts"
+
+function where(specs : any , testObj: any) {
```

Comment:
use type `Tests` in `whereAll` for type of `specs`.
Move `Tests` in `utils/types.ts` and change accordingly in `where`, `whereAll`, `whereAny`